### DISCUSSION TITLE
Unexpected outcome of the SaveImage transform

### DISCUSSION THREAD

**Discussion by hblanco2009** [Project-MONAI/MONAI#5361] [Q&A] [answered]
After loading and resizing a PNG image, the SaveImage transform doesn't work as expected. To the right you can see the original image (4280 x 3520 pixels). Below you can see the resized image (256 x 256) but clearly not well encoded. Any hints on what is going on?

![monai_issue_2](https://user-images.githubusercontent.com/61469912/196651619-f17b1391-c329-4f8f-add3-b17133f5517a.png)

**Comment by wyli** [comment, marked as answer]
thanks for reporting, could you please try `channel_dim=None` or `channel_dim=-1` for `EnsureChannelFirst`, because the input png may be channel last.

**Reply by hblanco2009** [reply]
Thank you for your soon reply. I did what you suggested but still it doesn't work. In fact, when the image is loaded the metadata information shows that the original image has no channel. 

![image_loaded](https://user-images.githubusercontent.com/61469912/196659054-3b4a5eca-154d-46f1-a807-b9a768f42dd7.png)

**Reply by hblanco2009** [reply]
This is the original image if you want to have a look at it: [https://drive.google.com/file/d/1jc3l85WY4NY5vKp3RA_A1dLoMYVT99nw/view?usp=sharing](url)

**Reply by wyli** [reply]
ok, ~then please try setting the `dtype` of `SaveImage`, perhaps `np.uint16`~

edit: should be `output_dtype=np.uint16`

**Reply by hblanco2009** [reply]
Nope, I tried several combinations: channel_dim: {'None', 'no_channel'}, dtype: {np.uint8, np.uint16, np.uint32}, but still the resized image is not well encoded.

**Reply by wyli** [reply]
thanks, if you could share one png example, I can look into this further

**Reply by hblanco2009** [reply]
This is the link: https://drive.google.com/file/d/1jc3l85WY4NY5vKp3RA_A1dLoMYVT99nw/view?usp=sharing
Let me know if you could download the image.

**Reply by wyli** [reply]
yes, it's `output_dtype=np.uint16` in `SaveImage`, it works fine. sorry the `dtype` was an incorrect arg

**Reply by hblanco2009** [reply]
Yes!, it works very good! Thank you. Cheers.
